app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -502,8 +502,10 @@ def clone_vm(raw_args)
       vm_model = new_vm(raw_args)
       opts['interfaces'] = vm_model.interfaces
       opts['volumes'] = vm_model.volumes
-      opts["customization_spec"] = client.cloudinit_to_customspec(args[:user_data]) if args[:user_data]
-      client.servers.get(client.vm_clone(opts)['new_vm']['id'])
+      opts["customization_spec"] = client.cloudinit_to_customspec(args[:user_data]) if args[:user_data] && valid_cloudinit_for_customspec?(args[:user_data])
```

Comment:
Nitpick: this line is quite long for modifier statement. Would prefer having this on multiple lines (I hope rubocop doesn't mind, otherwise, it's just silly :)